PurchasesCoreSwift/Logging/LoggerWrapper.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  LoggerWrapper.swift
+//  PurchasesCoreSwift
+//
+//  Created by Tina Nguyen on 12/08/20.
+//  Copyright © 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+@objc(RCLogIntent) public enum LogIntent: Int {
+    case appleError
+    case info
+    case purchase
+    case rcError
+    case rcPurchaseSuccess
+    case rcSuccess
+    case user
+    case warning
+    
+    func emojis() -> String {
+        switch self {
+        case .appleError: return Emojis.appleError.rawValue + Emojis.doubleExclamation.rawValue
+        case .info: return Emojis.info.rawValue
+        case .purchase: return Emojis.purchase.rawValue
+        case .rcError: return Emojis.rcError.rawValue + Emojis.doubleExclamation.rawValue
+        case .rcPurchaseSuccess: return Emojis.rcSuccess.rawValue + Emojis.purchase.rawValue
+        case .rcSuccess: return Emojis.rcSuccess.rawValue
+        case .user: return Emojis.appUserID.rawValue
+        case .warning: return Emojis.warning.rawValue
+        }
+    }
+}
+
+@objc(RCLoggerWrapper) public class LoggerWrapper: NSObject {
+    @objc public static func log(level: LogLevel, intent: LogIntent, message: String) {
+        let emojifiedMessage = "\(intent.emojis()) \(message)"
+        Logger.log(level: level, message: emojifiedMessage)
+    }
+    @objc public static func appleError(_ message: String) {
+        log(level: .error, intent: .appleError, message: message)
+    }
+    
+    @objc public static func appleInfo(_ message: String) {
+        log(level: .info, intent: .appleError, message: message)
+    }
+    
+    @objc public static func debugInfo(_ message: String) {
```

Comment:
Also following Android implementation. Examples: "setCreative called" and "Debug logging enabled"

Reply:
I think we might wanna just replace the `debug` method in `Logger` with this one, and we could keep the name `debug` so that there are no other code changes needed

Reply:
that way all existing calls to `RCDebugLog` automatically gain the prefix

Reply:
I was planning on merging this PR to work on the other string files, but then other developers would see the info emoji before releasing debug logs. Should this be refactored after all string files are merged?

Reply:
On second thought, I could create another branch, something like "[FINAL] debug logs" and merge all string files into that branch. Once we're ready for release merge that into develop. Thoughts?

Reply:
On second thought, I could create another branch called "[FINAL] debug logs" and merge all changes (this PR + strings file) into there. Once we're ready for release we would merge that branch into develop. Thoughts?

Reply:
if we're worried about sending out a half-implemented version of this to the world, perhaps we could use a feature flag in order to have the whole code easy to test for us but not actually used by customers. 
as in, 

```swift
@objc public static func log(level: LogLevel, intent: LogIntent, message: String) {
#if V2_LOGS_ENABLED
    let emojifiedMessage = "\(intent.emojis()) \(message)"
    Logger.log(level: level, message: emojifiedMessage)
#else 
    Logger.log(level: level, message: message)
#end
}
```

Reply:
I tried replacing the `debug` method in `Logger`, but `RCDebugLog` calls aren't going into the `log(level: LogLevel, intent: LogIntent, message: String)` method. I placed a breakpoint at `RCDebugLog(@"Debug logging enabled.");` in RCPurchases and clicking into the function brought me to the `log(level: LogLevel, message: String)` function. 

Also, where do I define the custom compiler flag? I tried adding it to the target's build settings under Preprocessor Macros and tried Swift Compiler - Custom Flags, but the `V2_LOGS_ENABLED` don't populate when I try a test call of `[RCLogger appleError: @"test"]`